### PR TITLE
Changed implementation of computeLiquidityValue, computeProfitMaximizingTrade, getAmountIn, getAmountOut to cater for variable swapFee and platformFee

### PR DESCRIPTION
**Pull request by xenide**
So I basically copied the implementation for _calcFee over to computeLiquidityValue like Oliver suggested. 

But the test cases involving platformFee are failing. The numbers are off by a little, in the 10e-6% kind of range. I went through all the relevant code a few times, but I think that my copied implementation should be correct. So I can't tell if the deviation shown in the test cases are a result of the accuracy scalings. 

In any case, I think I need more test cases to test the range of swapFee and platformFee.

### CODE REVIEW

**Comment by leodsoft** [comment]
I will take a closer look, but if you mean error against the original uniswap values, most likely yes this is just the difference in precision between our basis-point based decimal fee - essentially a percentage to 2 decimal places, versus the original uniswap constant fee which was calculated as a fraction (1 sixth)... 

When setting to close values this is the equivalent of 0.1667 and 0.1666666666666666666666 (repeater).

**Comment by leodsoft** [comment]
Note I just added some minor changes for some whitespace and aligning expected gas fees. Remaining failures are explicitly those as mentioned - the fee calc discrepancies, which are most likely just a question of accuracy.

On that, I also added the explanatory text around the testing strategy that I had added in Trello - so we have it there with the test-code to hopefully help others in the future.

File: uniswap-v2-periphery/contracts/examples/ExampleSwapToPrice.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/examples/ExampleSwapToPrice.sol]: Drop to newline to maintain Uniswap's styling? (They seem to just group tokens and reserves, so grouping swapFee is probably not what they intended)

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol]: Drop closing for `}` to newline

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol]: Drop closing `}` to newline

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Can probably remove the `ASSERT` comments

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Trailing whitespace

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (OliverNChalk) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Insert newline above `feeLiquidity` to emulate the styling in `V2Pair.sol` and make it more obvious they are equivalent

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (OliverNChalk) [inline uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Rename `sqrtNewK` -> `sqrtOldK`, and `rootKLast` -> `sqrtOldK` for consistency with V2Pair

**Review by OliverNChalk — COMMENTED** [review]
Some minor linting and style feedback, changes look good though. Will defer to Ian on the fee testing queries as it seems he's already responded there.

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol]: Fixed. Somehow I didn't see that at all, probably because I have like 2 columns on my screen. Thanks for catching

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2Library.sol]: Fixed

File: uniswap-v2-periphery/contracts/examples/ExampleSwapToPrice.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/examples/ExampleSwapToPrice.sol]: Fixed

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Fixed

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Fixed

File: uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol
Comment (xenide) [reply uniswap-v2-periphery/contracts/libraries/VexchangeV2LiquidityMathLibrary.sol]: Fixed

**Review by OliverNChalk — APPROVED** [review]
Looks good, let's hold off merging until the audtiors get back to us. We can chat about this in our call

**Comment by xenide** [comment]
Just fixed the expected values in the tests. All test passed.

**Review by leodsoft — APPROVED** [review]
Looks good. I added a cleanup of a couple of old comments in pair re the fee representation as well.

**Pull request merged**
